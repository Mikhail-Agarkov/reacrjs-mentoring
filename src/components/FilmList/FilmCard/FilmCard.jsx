import React from 'react';
import styles from './filmcard.module.scss';

const FilmCard = (props) => {
    const {poster_path, title, release_date, genres, id} = props['film'];
    return (
        <div className={styles.content} key={id}>
            <img src={poster_path} alt={title} className={styles.cover}/>
            <div className={styles.info}>
                <span>{title}</span>
                <span className={styles.date}>{getYearFromDateString(release_date)}</span>
            </div>
            <span className={styles.genre}>{genresToString(genres)} </span>
        </div>
    );
};

function genresToString(genres) {
    return genres.join(', ');
}

function getYearFromDateString(date) {
    return date.split('-')[0];
}

export default FilmCard;
