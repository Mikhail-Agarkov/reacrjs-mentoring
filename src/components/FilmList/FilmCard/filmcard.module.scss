.content {
  padding: 0 20px;
  width: 324px;
  height: 540px;
  font-family: Montserrat;
  color: #fff;
  cursor: pointer;
}

.cover {
  height: 455px;
  width: 324px;
}

.info {
  display: flex;
  justify-content: space-between;
  align-items: baseline;
  margin-top: 8px;
  font-size: 18px;
  opacity: 0.9;

  .date {
    border: 1px solid #979797;
    border-radius: 4px;
    padding: 2px 6px;
  }
}

.genre {
  font-size: 14px;
  opacity: 0.5;
}