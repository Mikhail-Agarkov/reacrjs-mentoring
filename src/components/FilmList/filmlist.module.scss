.content {
  min-height: 273px;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  padding: 60px 20px;
  background-color: #232323;
}

.notFound {
  color: #FFFFFF;
  font-family: Montserrat, sans-serif;
  font-size: 20px;
  text-transform: uppercase;
  line-height: 273px;
}