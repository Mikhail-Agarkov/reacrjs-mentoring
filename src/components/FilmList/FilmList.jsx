import React from 'react';
import FilmCard from './FilmCard/FilmCard';
import styles from './filmlist.module.scss';

const FilmList = (props) => (
    <section className={styles.content}>
        {props.films.map(film => <FilmCard film={film}/>)}
    </section>
);

export default FilmList;
