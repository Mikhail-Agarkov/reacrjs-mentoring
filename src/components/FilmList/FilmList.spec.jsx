import React from 'react';
import {shallow} from 'enzyme';

import FilmList from "./FilmList";
import FilmCard from "./FilmCard/FilmCard";

describe('FilmCard', () => {
    it('should render film list section correctly', () => {
        const component = shallow(<FilmList films={
            [
                {
                    'src': 'src/to/image',
                    'alt': 'image alt text',
                    'title': 'Film Title',
                    'date': '2019',
                    'genre': 'Fantasy',
                    'key': '2516'
                },
                {
                    'src': 'src/to/another/image',
                    'alt': 'another image alt text',
                    'title': 'Another Film Title',
                    'date': '2020',
                    'genre': 'Comedy',
                    'key': '6498'
                }
            ]
        }/>);
        expect(component).toMatchSnapshot();
    })
});