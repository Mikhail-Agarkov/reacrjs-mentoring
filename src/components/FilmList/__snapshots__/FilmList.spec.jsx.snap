// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCard should render with empty Films state from Redux store 1`] = `
<section
  className="content"
>
  <div
    className="notFound"
  >
    No films found.
  </div>
</section>
`;

exports[`FilmCard should render with given state from Redux store 1`] = `
<section
  className="content"
>
  <div
    className="content"
  >
    <img
      alt="Film Title"
      className="cover"
      src="src/to/image"
    />
    <div
      className="info"
    >
      <span>
        Film Title
      </span>
      <span
        className="date"
      >
        2019
      </span>
    </div>
    <span
      className="genre"
    >
      Fantasy
       
    </span>
  </div>
  <div
    className="content"
  >
    <img
      alt="Another Film Title"
      className="cover"
      src="src/to/another/image"
    />
    <div
      className="info"
    >
      <span>
        Another Film Title
      </span>
      <span
        className="date"
      >
        2020
      </span>
    </div>
    <span
      className="genre"
    >
      Comedy
       
    </span>
  </div>
</section>
`;

exports[`FilmCard should render with undefined Films state from Redux store 1`] = `
<section
  className="content"
>
  <div
    className="notFound"
  >
    No films found.
  </div>
</section>
`;
